neurodamus/core/configuration.py

```diff
@@ -298,6 +296,69 @@ def init(cls, config_file, cli_options):
         logging.info("Initializing hoc config objects")
         cls._init_hoc_config_objs()
 
+    @classmethod
+    def save_path(cls):
+        """Default to output_root if none is provided"""
+        return cls.save or cls.output_root
+
+    @classmethod
+    def coreneuron_datadir_save(cls):
+        """Default to output_root if none is provided"""
+        return cls.coreneuron_datadir or str(Path(cls.save_path()) / "coreneuron_input")
+
+    @classmethod
+    def coreneuron_datadir_restore(cls):
+        """Default to output_root if none is provided"""
+        return str(Path(cls.restore) / "coreneuron_input")
```

Comment:
Another thought: is there a need to have 2 additional properties `coreneuron_datadir_save` and `coreneuron_datadir_restore` ? Can we just assign different values to `coreneuron_datadir` in case of save and restore, and use it in the `node.py` ?

Reply:
to me `coreneuron_datadir` shouldn't even exist. I have that because `_coreneuron_configure_datadir` may override it in case of SHM setup. It is true that `_setup_coreneuron_restore_environment` is the only one that uses the restore one and only to link it to save so I could also just move the logic there. However, I kept it in this way to follow the same logic as the rest. I do need to link because coreneuron cannot have 2 distinct folders

Reply:
to distinguish it from coreneuron_datadir_restore_path. I split everything in save/restore so it is clear where we are going to point to. The fact that coreneuron does not allow for different folders is another one

Reply:
> to distinguish it from coreneuron_datadir_restore_path. I split everything in save/restore so it is clear where we are going to point to. The fact that coreneuron does not allow for different folders is another one

How about the no save/no restore case which is the usecase most of the time ? Do you mean that we will always use `coreneuron_datadir_save`?  I feel that's odd.

Reply:
ok, I see that we do not agree on this point. I yield. 1 folder to rule them all and add-hoc code to link restore to save